mlir-sycl/test/Conversion/SYCLToLLVM/sycl-methods-to-llvm.mlir

```diff
@@ -1,121 +1,295 @@
 // RUN: sycl-mlir-opt -split-input-file -convert-sycl-to-llvm="use-bare-ptr-call-conv" %s | FileCheck %s
-// XFAIL: *
 
 //===-------------------------------------------------------------------------------------------------===//
-// sycl.nd_range.get_global_range
+// sycl.range.get with scalar result type
 //===-------------------------------------------------------------------------------------------------===//
 
-!sycl_id_3_ = !sycl.id<[3], (!sycl.array<[3], (memref<3xi64, 4>)>)>
 !sycl_range_3_ = !sycl.range<[3], (!sycl.array<[3], (memref<3xi64, 4>)>)>
-!sycl_nd_range_3_ = !sycl.nd_range<[3], (!sycl_range_3_, !sycl_range_3_, !sycl_id_3_)>
 
 // CHECK-LABEL:   llvm.func @test(
-// CHECK-SAME:                    %[[VAL_0:.*]]: !llvm.struct<"class.sycl::_V1::nd_range.3", (struct<"class.sycl::_V1::range.3", (struct<"class.sycl::_V1::detail::array.3", (array<3 x i64>)>)>, struct<"class.sycl::_V1::range.3", (struct<"class.sycl::_V1::detail::array.3", (array<3 x i64>)>)>, struct<"class.sycl::_V1::id.3", (struct<"class.sycl::_V1::detail::array.3", (array<3 x i64>)>)>)>) -> !llvm.struct<"class.sycl::_V1::range.3", (struct<"class.sycl::_V1::detail::array.3", (array<3 x i64>)>)> {
-// CHECK-NEXT:      %[[VAL_1:.*]] = llvm.extractvalue %[[VAL_0]][0] : !llvm.struct<"class.sycl::_V1::nd_range.3", (struct<"class.sycl::_V1::range.3", (struct<"class.sycl::_V1::detail::array.3", (array<3 x i64>)>)>, struct<"class.sycl::_V1::range.3", (struct<"class.sycl::_V1::detail::array.3", (array<3 x i64>)>)>, struct<"class.sycl::_V1::id.3", (struct<"class.sycl::_V1::detail::array.3", (array<3 x i64>)>)>)>
-// CHECK-NEXT:      llvm.return %[[VAL_1]] : !llvm.struct<"class.sycl::_V1::range.3", (struct<"class.sycl::_V1::detail::array.3", (array<3 x i64>)>)>
+// CHECK-SAME:                    %[[VAL_0:.*]]: !llvm.ptr<[[RANGE3:.*]]>,
```

Comment:
> This some pretty advanced use of C++ ! I like the fact the code generation scheme in this PR allows reduction of code duplication, but I wonder whether the added complexity in the code is worth it. Feel like I have to brush up my C++ after looking at this code :-)

We will be lowering +20 operations in a **very** similar fashion (I got tired of fixing small issues in +10 places each time). I think it'll be worth it long-term. Also, this will ease adding new ops with a similar lowering and even adding lowering for other targets!

Reply:
I did not want to overcomplicate this file... We already have a file to check we're lowering types correctly. As we're only checking operations lowering here, I'd like to keep as is (cleaner, also). We'll have deeply nested types and the file will become unreadable pretty quickly. If you think that's ok, we can keep type lowering tests in the other file.